examples/albef/finetune.py

```diff
@@ -0,0 +1,198 @@
+                ans_lengths=ans_lengths,
+                alpha=alpha,
+                is_train=True,
+            )
+
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+            if epoch == 0 and batch % step_size == 0 and batch <= warmup_iterations:
+                scheduler.step(batch // step_size)
+
+            if batch % log_every_n_steps == 0:
+                total_time = time.time() - start_time
+                time_str = "time {},".format(
+                    datetime.timedelta(seconds=int(total_time))
+                )
+                epoch_str = "epoch {}/{},".format(epoch, args["max_epochs"])
+                batch_str = "batch {}/{},".format(batch, len(data_loader))
+                loss_str = "loss {}".format(loss.item())
+                print(time_str, epoch_str, batch_str, loss_str)
+
+        save_obj = {
+            "model": model_without_ddp.state_dict(),
+            "optimizer": optimizer.state_dict(),
+            "lr_scheduler": scheduler.state_dict(),
+            "epoch": epoch,
+        }
+        torch.save(
+            save_obj, os.path.join(checkpoint_root, "vqa_checkpoint_%02d.pth" % epoch)
+        )
+
+
+@torch.no_grad()
+def evaluation(model, datamodule, k, device):
+    model.eval()
+
+    result = []
+
+    answer_list = datamodule.test_dataset.answer_list
+    answer_input_ids = datamodule.test_dataset.answer_input_ids.to(device)
+    answer_atts = datamodule.test_dataset.answer_attention_mask.to(device)
+    data_loader = datamodule.test_dataloader(
+        is_distributed=dist_utils.is_dist_avail_and_initialized(),
+        num_tasks=dist_utils.get_world_size(),
+        global_rank=dist_utils.get_rank(),
+    )
+
+    start_time = time.time()
+    log_every_n_steps = 100
+
+    for batch, (img, ques, ques_atts, ques_ids) in enumerate(data_loader):
+        img = img.to(device, non_blocking=True)
+        ques = ques.to(device)
+        ques_atts = ques_atts.to(device)
+
+        topk_ids, topk_probs = model(
+            img, ques, ques_atts, answer_input_ids, answer_atts, k=k, is_train=False
+        )
+
+        for ques_id, topk_id, topk_prob in zip(ques_ids, topk_ids, topk_probs):
+            _, pred = topk_prob.max(dim=0)
+            result.append(
+                {"question_id": ques_id, "answer": answer_list[topk_id[pred]]}
+            )
+
+        if batch % log_every_n_steps == 0:
+            total_time = time.time() - start_time
+            total_time_str = str(datetime.timedelta(seconds=int(total_time)))
+            print(
+                "time {}, batch {}/{}".format(total_time_str, batch, len(data_loader))
+            )
+
+    return result
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", default="./examples/albef/configs/vqa.yaml")
+    parser.add_argument("--device", default="cuda")
+    parser.add_argument("--distributed", default=True, type=bool)
```

Comment:
fyi I found [this warning](https://docs.python.org/3/library/argparse.html#type:~:text=The%20bool()%20function%20is%20not%20recommended%20as%20a%20type%20converter.%20All%20it%20does%20is%20convert%20empty%20strings%20to%20False%20and%20non%2Dempty%20strings%20to%20True.%20This%20is%20usually%20not%20what%20is%20desired.) against using type=bool. An alternative is to use `action='store_true'` or `action='store_false'`

Reply:
Thanks for the info! I actually reset this arg in a separate method so I removed it from the command line args.